lib/spdy-transport/protocol/http2/parser.js

```diff
@@ -386,6 +419,28 @@ Parser.prototype.onSettingsFrame = function onSettingsFrame(header,
       settings[name] = value;
   }
 
+  if (constants.settingsIndex[constants.settings_server.SETTINGS_ENABLE_PUSH] in settings) {
```

Comment:
This is absolutely fine. We could as well say that `SETTINGS_ENABLE_PUSH` may be changed, and it will be hard to refactor every place that is using it ;)

The idea behind `settingsIndex` is that it is very convenient to use in the protocol parser, it should not really be used anywhere else. This PR's use case doesn't really look that convenient.